lib/QuipProcessor.js

```diff
@@ -295,8 +295,27 @@ class QuipProcessor {
                 dateStr = updatedDate === createdDate? createdDate : `${createdDate} (Updated: ${updatedDate})`;
                 dateStr = `, ${dateStr}`;
             }
+
+            let referencedSections = "";
+            if(message.annotation && message.annotation.id) {
+                if(message.annotation.highlight_section_ids) {
+                    referencedSections = message.annotation.highlight_section_ids.map((sectionId, index) => {
+                        // Not sure if there can be more than one highlight section,
+                        // but if so, we can number the reference links
+                        const suffix =
+                            message.annotation.highlight_section_ids.length == 1
+                            ? ""
+                            : ` ${index + 1}`;
+                        return `<a href="#${sectionId}">ref${suffix}</a>`;
```

Comment:
I picked an arbitrary string to use for the reference links, it is either just "ref" or "ref 1", "ref 2" etc if there happen to be multiple values in highlight_section_ids. I don't yet know if this happens in practice but I figured I'd handle the case anyway. Open to any other suggestions on how to name these links.

I also considered just making the timestamp a link but decided against that.